File: src/main/scala/com/vc/medium/L701.scala
package com.vc.medium

class TreeNode701(var _value: Int) {
  var value: Int = _value
  var left: TreeNode701 = null
  var right: TreeNode701 = null
}

object Solution {
  def insertIntoBST(root: TreeNode701, value: Int): TreeNode701 = {
    var current = root
    var prev = current
    while(current != null) {
      prev = current
      if(current.value > value)
        current = current.left
      else
        current = current.right
    }

    if(prev.value > value) prev.left = new TreeNode701(value)
    else prev.right = new TreeNode701(value)

    root
  }
}

File: src/main/java/com/vc/medium/L701.java
package com.vc.medium;

import com.vc.hard.TreeNode;

class L701 {
    public TreeNode insertIntoBST(TreeNode root, int val) {
        TreeNode current = root, prev = root;
        while(current != null) {
            if(current.val > val) {
                prev = current;
                current = current.left;
            }
            else if(current.val < val) {
                prev = current;
                current = current.right;
            }
            else return root;
        }

        if(prev.val > val) prev.left = new TreeNode(val);
        else prev.right = new TreeNode(val);
        return root;
    }
}
